kong/plugins/moesif/handler.lua

```diff
@@ -59,7 +59,7 @@ function MoesifLogHandler:access(conf)
   if (queue_hashes[hash_key] == nil) or 
         (queue_hashes[hash_key] ~= nil and type(queue_hashes[hash_key]) == "table" and #queue_hashes[hash_key] < conf.event_queue_size) then
 
-    if (content_length == nil) or (tonumber(content_length) <= conf.max_body_size_limit) then 
+    if (content_length == nil and string.len(req_get_body_data()) <= conf.max_body_size_limit) or (content_length ~= nil and tonumber(content_length) <= conf.max_body_size_limit) then 
```

Comment:
this will throw exception when req_get_body_data returns nil. May be store req_get_body_data in some variable and check if its not nil